### ISSUE TITLE
Enhancement: Add negative number patterns to no-magic-numbers ignore

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#34]
## Description

The `no-magic-numbers` rule ignores `[0, 1, -1, 2]` by default, but negative numbers like `-1` are handled awkwardly. The current implementation checks for `UnaryExpression` with `-` operator separately.

## Suggested Enhancement

Add an option like `ignoreNegated: true` that automatically ignores the negated versions of any number in the `ignore` list.

## Use Case

```ts
// Common patterns that might be flagged unnecessarily
const NOT_FOUND = -1;
if (result === -1) return;
arr.at(-1); // last element
```

Currently users have to add `-1` explicitly to ignore list. With `ignoreNegated: true`, adding `1` would also allow `-1`.
